Content.Server/Resist/ResistLockerSystem.cs

```diff
@@ -0,0 +1,117 @@
+using Content.Server.Popups;
+using Robust.Shared.Localization;
+
+namespace Content.Server.Resist;
+
+public class ResistLockerSystem : EntitySystem
+{
+    [Dependency] private readonly DoAfterSystem _doAfterSystem = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+    [Dependency] private readonly LockSystem _lockSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<ResistLockerComponent, RelayMovementEntityEvent>(OnRelayMovement);
+        SubscribeLocalEvent<ResistLockerComponent, ResistDoAfterComplete>(OnDoAfterComplete);
+        SubscribeLocalEvent<ResistLockerComponent, ResistDoAfterCancelled>(OnDoAfterCancelled);
+        SubscribeLocalEvent<ResistLockerComponent, EntRemovedFromContainerMessage>(OnRemovedFromContainer);
+    }
+
+    private void OnRelayMovement(EntityUid uid, ResistLockerComponent component, RelayMovementEntityEvent args)
+    {
+        if (!TryComp(uid, out EntityStorageComponent? storageComponent))
+            return;
+
+        if (!component.IsResisting)
+        {
+            if (TryComp<LockComponent>(uid, out var lockComponent) && lockComponent.Locked || storageComponent.IsWeldedShut)
+            {
+                AttemptResist(args.Entity, uid, storageComponent, component);
+            }
+        }
+    }
+
+    private void AttemptResist(EntityUid user, EntityUid target, EntityStorageComponent? storageComponent = null, ResistLockerComponent? resistLockerComponent = null)
+    {
+        if (!Resolve(target, ref storageComponent, ref resistLockerComponent))
+            return;
+
+        resistLockerComponent.CancelToken = new();
+        var doAfterEventArgs = new DoAfterEventArgs(user, resistLockerComponent.ResistTime, resistLockerComponent.CancelToken.Token, target)
+        {
+            BreakOnTargetMove = false,
+            BreakOnUserMove = true,
+            BreakOnDamage = true,
+            BreakOnStun = true,
+            NeedHand = false, //No hands 'cause we be kickin'
+            TargetFinishedEvent = new ResistDoAfterComplete(user, target),
+            TargetCancelledEvent = new ResistDoAfterCancelled(user)
+        };
+
+        resistLockerComponent.IsResisting = true;
+        _popupSystem.PopupEntity(Loc.GetString("resist-locker-component-start-resisting"), user, Filter.Entities(user));
+        _doAfterSystem.DoAfter(doAfterEventArgs);
+    }
+
+    private void OnDoAfterComplete(EntityUid uid, ResistLockerComponent component, ResistDoAfterComplete ev)
+    {
+        component.IsResisting = false;
+
+        if (TryComp<EntityStorageComponent>(uid, out var storageComponent))
+        {
+            if (storageComponent.IsWeldedShut)
+                storageComponent.IsWeldedShut = false;
+
+            if (TryComp<LockComponent>(ev.Target, out var lockComponent))
+            {
+                _lockSystem.Unlock(uid, ev.User, lockComponent);
+            }
```

Comment:
```suggestion
                _lockSystem.Unlock(uid, ev.User, lockComponent);
```
For consistency with other single line if statement.